File: 프로그래머스/0/181882. 조건에 맞게 수열 변환하기 1/조건에 맞게 수열 변환하기 1.java
class Solution {
    public int[] solution(int[] arr) {
        int[] answer = new int[arr.length];
        
        for(int i =0; i<arr.length; i++){
            int target = arr[i];
            if(target >= 50 && target % 2 == 0) answer[i] = target/2;
            else if(target < 50 && target % 2 != 0) answer[i] = target*2;
            else answer[i] = target;
        }
        return answer;
    }
}

File: 프로그래머스/0/181882. 조건에 맞게 수열 변환하기 1/조건에 맞게 수열 변환하기 1.kt
class Solution {
    fun solution(arr: IntArray): IntArray = arr.map{
        if((it >= 50) and (it % 2 == 0)){
                it / 2
        } else if((it < 50) and (it % 2 != 0)){
                it*2
        }else{
                it
        }
    }.toIntArray()
}